dashboard/test/models/user_test.rb

```diff
@@ -2359,6 +2359,24 @@ def email_preference_params(**args)
     assert_nil user.parent_email
   end
 
+  test 'upgrade_to_teacher given valid params should delete family_name property' do
+    DCDO.stubs(:get).with('family-name-features', false).returns(true)
+
+    family_name = 'TestFamName'
+    user = User.create(@good_data)
+    user.properties = {family_name: family_name}
+    user.save!
```

Comment:
I'm not sure if that would work because the family_name is being stored in the `properties` blob. It _might_ work if `family_name` is added to `serialized_attrs` but even then I'm not sure 🤔

Reply:
it should still be a single db write though, activerecord runs queries lazily and won't execute anything until the record is actually saved.

Reply:
`create` isn't lazy, it's essentially the same as `new` plus `save`.

> Creates an object (or multiple objects) and saves it to the database, if validations pass. ([apidock](https://apidock.com/rails/ActiveRecord/Base/create/class))

Reply:
ohh I see, I thought you meant the `.properties =` part. i think it should be possible to get this down to a single query

Reply:
Ah, yep! The properties line definitely isn't saving.

Reply:
shortened this in e11fdde07ca6cad5951d875321277ef291d0284d. I noticed this is in the test, not the model itself; is this kind of query efficiency in tests something we generally prioritize?

Reply:
Tests are pretty slow, so I think easy opportunities to speed them up without significantly sacrificing readability should be taken.

I suspect we have a lot of test of the form `test that a function behaves properly given a saved object with a specific property`, so little inefficiencies can add up.